src/fu-main.c

```diff
@@ -1662,6 +1670,16 @@ fu_main_load_introspection (const gchar *filename, GError **error)
 	return g_dbus_node_info_new_for_xml (g_bytes_get_data (data, NULL), error);
 }
 
+static gboolean
+fu_main_is_hypervisor (void)
+{
+	g_autofree gchar *buf = NULL;
+	gsize bufsz = 0;
+	if (!g_file_get_contents ("/proc/cpuinfo", &buf, &bufsz, NULL))
+		return FALSE;
+	return g_strstr_len (buf, (gssize) bufsz, "hypervisor") != NULL;
```

Comment:
Is this valid for all VMs?  Or does there need to be more ways to detect?

Reply:
I think the kernel has a pretty long list of ways to detect a VM.